core/src/com/unciv/ui/screens/diplomacyscreen/TradeTable.kt

```diff
@@ -49,6 +51,28 @@ class TradeTable(
                 retractOffer()
                 return@onClick
             }
+            // If there is a research agreement trade, make sure both civilizations should be able to pay for it.
+            // If not lets add an extra gold offer to satisfy this.
+            // There must be enough gold to add to the offer to satisfy this, otherwise the research agreement button would be disabled
+            if (tradeLogic.currentTrade.ourOffers.any { it.name == Constants.researchAgreement}) {
+                val researchCost = currentPlayerCiv.diplomacyFunctions.getResearchAgreementCost()
+                val currentPlayerOfferedGold = tradeLogic.currentTrade.ourOffers.firstOrNull { it.type == TradeType.Gold }?.amount ?: 0
+                val otherCivOfferedGold = tradeLogic.currentTrade.theirOffers.firstOrNull { it.type == TradeType.Gold }?.amount ?: 0
+                // Check if we require more gold from them
+                if (currentPlayerCiv.gold + otherCivOfferedGold < researchCost) {
```

Comment:
Here I see that the sum of both player's gold needs to be the research cost, whereas above in OffersListScroll it's twice the research cost - seems inconsistent?

Reply:
In OffersListScroll we are checking if both civilization's gold combined can pay for the total research cost. Here in TradeTable we are checking how much gold one civilization needs to give the other. We can assume that both of them combined can pay for it since otherwise the button would not be pressable.

Just a note: The AI will only offer the research agreement when both sides can pay for their half of the agreement.